test/e2e_node/docker_test.go

```diff
@@ -70,4 +77,174 @@ var _ = framework.KubeDescribe("Docker features [Feature:Docker]", func() {
 			}
 		})
 	})
+
+	Context("when live-restore is enabled", func() {
+		It("containers should not be disrupted when the daemon is down [ygg]", func() {
+			const (
+				startTimeFilename   = "start-timestamp"
+				currentTimeFilename = "current-timestamp"
+			)
+			isSupported, err := isDockerLiveRestoreSupported()
+			framework.ExpectNoError(err)
+			if !isSupported {
+				framework.Skipf("Docker live-restore is not supported.")
+			}
+
+			By("Check whether live-restore is enabled.")
+			isEnabled, err := isDockerLiveRestoreEnabled()
+			framework.ExpectNoError(err)
+			framework.Logf("Docker live-restore enabled: %t", isEnabled)
+			if !isEnabled {
+				// Enables Docker live-restore if it's supported but disabled.
+				By("Enable Docker live-restore.")
+				framework.ExpectNoError(setDockerLiveRestore(true))
+
+				b, err := isDockerLiveRestoreEnabled()
+				framework.ExpectNoError(err)
+				if !b {
+					framework.Logf("Failed to enable Docker live-restore: %v", err)
+				}
+
+				defer func() {
```

Comment:
I tried to panic in the defer function and the cleanup worked as expected, so the ginkgo test framework handles this for us. :)